lib/routes.js

```diff
@@ -92,26 +93,38 @@ function fastifySwagger (fastify, opts, done) {
     }
   })
 
+  const hasTransformSwaggerFn = typeof opts.transformSwagger === 'function'
+  const transformSwagger = opts.transformSwagger
   fastify.route({
     url: '/json',
     method: 'GET',
     schema: { hide: true },
     ...hooks,
-    handler: function (req, reply) {
-      reply.send(fastify.swagger())
-    }
+    handler: hasTransformSwaggerFn
+      ? function (req, reply) {
+        reply.send(transformSwagger(req, reply, JSON.parse(JSON.stringify(fastify.swagger()))))
+      }
+      : function (req, reply) {
+        reply.send(fastify.swagger())
+      }
   })
 
   fastify.route({
     url: '/yaml',
     method: 'GET',
     schema: { hide: true },
     ...hooks,
-    handler: function (req, reply) {
-      reply
-        .type('application/x-yaml')
-        .send(fastify.swagger({ yaml: true }))
-    }
+    handler: hasTransformSwaggerFn
+      ? function (req, reply) {
+        reply
+          .type('application/x-yaml')
+          .send(yaml.stringify(transformSwagger(req, reply, JSON.parse(JSON.stringify(fastify.swagger())))))
```

Comment:
I do JSON.parse(JSON.stringify)) to get an immutable object. I dont think that this is smart.